src/order/jordan_hoelder.lean

```diff
@@ -0,0 +1,803 @@
+/-
+Copyright (c) 2021 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes
+-/
+import order.lattice
+import data.fin
+import data.set_like.basic
+import data.list.sort
+import data.equiv.fin
+import data.equiv.functor
+/-!
+# Jordan Hölder Theorem
+
+This file proves the Jordan Hölder theorem for a `jordan_hoelder_lattice`, a class also defined in
+this file. Examples of `jordan_hoelder_lattice` include `subgroup G` if `G` is a group, and
+`submodule R M` if `M` is an `R`-module. Using this approach the theorem need not be proved
+seperately for both groups and modules, the proof in this file can be applied to both.
+
```

Comment:
I think there needs to be an explanation or TODO stating that this PR doesn't yet provide the instances for groups or modules.